app/controllers/teammates_controller.rb

```diff
@@ -0,0 +1,60 @@
+class TeammatesController < ApplicationController
+  before_action :require_invitation
+  before_action :require_pending
+  helper_method :already_teammate
+  rescue_from ActiveRecord::RecordNotFound, :with => :incorrect_token
+
+  def accept
+    if current_user && !already_teammate?
```

Comment:
I believe it is necessary - the first one is to make sure that if we have a logged in user they are not already a teammate for the event (so a user can't end up being multiple teammates for the same event) - if they are not a teammate, the invitation gets accepted. The second check is to set a flash and redirect if the logged in user is already a teammate and has tried to accept a second invitation.